src/hotspot/share/gc/shenandoah/shenandoahUtils.cpp

```diff
@@ -111,17 +111,20 @@ ShenandoahConcurrentPhase::~ShenandoahConcurrentPhase() {
   _timer->register_gc_concurrent_end();
 }
 
-ShenandoahTimingsTracker::ShenandoahTimingsTracker(ShenandoahPhaseTimings::Phase phase) :
+ShenandoahTimingsTracker::ShenandoahTimingsTracker(ShenandoahPhaseTimings::Phase phase, bool should_aggregate_cycles) :
   _timings(ShenandoahHeap::heap()->phase_timings()), _phase(phase) {
   assert(Thread::current()->is_VM_thread() || Thread::current()->is_ConcurrentGC_thread(),
           "Must be set by these threads");
   _parent_phase = _current_phase;
   _current_phase = phase;
   _start = os::elapsedTime();
+  _should_aggregate_cycles = should_aggregate_cycles;
 }
 
 ShenandoahTimingsTracker::~ShenandoahTimingsTracker() {
-  _timings->record_phase_time(_phase, os::elapsedTime() - _start);
+  const double end_time = os::elapsedTime();
+  const double phase_elapsed_time = end_time - _start;
+  _timings->record_phase_time(_phase, phase_elapsed_time, _should_aggregate_cycles);
```

Comment:
No need to introduce local variables here, right? The expression can stay inlined.